### ISSUE TITLE
[FreeBSD] user state option `empty_password: True` fails with Traceback

### ISSUE THREAD

**Issue by cedwards** [saltstack/salt#28382]
I'm using the following state and options to manage my user. I wanted to define a passwordless account and use ssh keys only, but the state fails with a traceback.

```
cedwards:
  user.present:
    - empty_password: True
    - groups:
      - wheel
```

```
saturn states (master*) # salt-call state.sls users.cedwards
[INFO    ] Loading fresh modules for state activity
[INFO    ] Fetching file from saltenv 'base', ** done ** u'users/cedwards.sls'
[INFO    ] Running state [cedwards] at time 15:42:42.827724
[INFO    ] Executing state user.present for cedwards
[INFO    ] Executing command '/sbin/zfs -?' in directory '/root'
[ERROR   ] An exception occurred in this state: Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/salt/state.py", line 1591, in call
    **cdata['kwargs'])
  File "/usr/local/lib/python2.7/site-packages/salt/states/user.py", line 405, in present
    __salt__['shadow.del_password'](name)
  File "/usr/local/lib/python2.7/site-packages/salt/utils/lazy.py", line 93, in __getitem__
    raise KeyError(key)
KeyError: 'shadow.del_password'

[INFO    ] Completed state [cedwards] at time 15:42:43.656104
local:
----------
          ID: cedwards
    Function: user.present
      Result: False
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "/usr/local/lib/python2.7/site-packages/salt/state.py", line 1591, in call
                  **cdata['kwargs'])
                File "/usr/local/lib/python2.7/site-packages/salt/states/user.py", line 405, in present
                  __salt__['shadow.del_password'](name)
                File "/usr/local/lib/python2.7/site-packages/salt/utils/lazy.py", line 93, in __getitem__
                  raise KeyError(key)
              KeyError: 'shadow.del_password'
     Started: 15:42:42.827724
    Duration: 828.38 ms
     Changes:

Summary for local
------------
Succeeded: 0
Failed:    1
------------
Total states run:     1
Total run time: 828.380 ms
```

**Comment by rallytime** [comment]
Thanks for the report and for fixing this up so quickly! Since your PR has been merged, is this issue good to close?

**Comment by cedwards** [comment]
I hadn't seen that it was merged yet. Yes, closing.

**Comment by rallytime** [comment]
Oh no problem, just making sure there wasn't more work to be done. Thanks!

**Comment by cedwards** [comment]
Also, you'll notice I made the assumption of 2015.8.3 in the docstring. I don't know if that needs to be changed..

**Comment by rallytime** [comment]
@cedwards I fixed it in the back-port to 2015.8 in #28404, which should merge forward and have 2015.8.2 be the correct release in both branches. :)
